diff -ruN tools/nrf5_sdk/components/libraries/fds/fds.c tools/nrf5_sdk_patched/components/libraries/fds/fds.c
--- tools/nrf5_sdk/components/libraries/fds/fds.c	2019-02-14 17:24:58.000000000 +0100
+++ tools/nrf5_sdk_patched/components/libraries/fds/fds.c	2019-10-14 14:50:07.814187172 +0200
@@ -1626,7 +1626,7 @@
 }
 
 
-static uint32_t flash_end_addr(void)
+uint32_t fds_flash_end_addr(void)
 {
     uint32_t const bootloader_addr = BOOTLOADER_ADDRESS;
     uint32_t const page_sz         = NRF_FICR->CODEPAGESIZE;
@@ -1649,7 +1649,7 @@
 static void flash_bounds_set(void)
 {
     uint32_t flash_size  = (FDS_PHY_PAGES * FDS_PHY_PAGE_SIZE * sizeof(uint32_t));
-    m_fs.end_addr   = flash_end_addr();
+    m_fs.end_addr   = fds_flash_end_addr();
     m_fs.start_addr = m_fs.end_addr - flash_size;
 }
 
diff -ruN tools/nrf5_sdk/components/libraries/fds/fds.h tools/nrf5_sdk_patched/components/libraries/fds/fds.h
--- tools/nrf5_sdk/components/libraries/fds/fds.h	2019-02-14 17:24:58.000000000 +0100
+++ tools/nrf5_sdk_patched/components/libraries/fds/fds.h	2019-10-14 14:50:16.506174804 +0200
@@ -693,6 +693,8 @@
 ret_code_t fds_stat(fds_stat_t * p_stat);
 
 
+uint32_t fds_flash_end_addr(void);
+
 /** @} */
 
 
